src/modm/platform/extint/rp/int_handler.hpp.in

```diff
@@ -0,0 +1,133 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include <modm/architecture/interface/interrupt.hpp>
+#include <modm/platform/gpio/base.hpp>
+#include <modm/platform/irq/int_controller.hpp>
+#include <modm/utils/inplace_function.hpp>
+
+namespace modm::platform
+{
+%% if with_gpio
+MODM_ISR_DECL(IO_IRQ_BANK0);
+%%endif
+%% if with_qspi
+MODM_ISR_DECL(IO_IRQ_QSPI);
+%%endif
+
+/**
+ * Interrupt Handler
+ *
+ * @ingroup modm_platform_irq
+ */
+template<IRQn_Type Type, size_t NLines>
+class IntHandler
+{
+public:
+	static constexpr size_t Lines = NLines;
+	using Handler =
+		modm::inplace_function<void(Gpio::InputTrigger), {{handler_storage}}, alignof(void*)>;
+
+public:
+	static void
+	enable(IntPriority priority = IntPriority::Default)
+	{
+		if (priority != IntController::getPriority(Type))
+		{
+			IntController::setPriority(Type, priority);
+		}
+		IntController::enable(Type);
+	}
+
+	static void
+	disable()
+	{
+		IntController::disable(Type);
+	}
+
+	template<class Pin>
+	static void
+	connect(Gpio::InputTrigger triggers, Handler&& handler)
+	{
+		static_assert(Pin::port == gpioPort());
+
+		disableInterrupts<Pin>(Gpio::InputTrigger::All);
+		acknowledgeInterrupts<Pin>(Gpio::InputTrigger::All);
+
+		handlers[Pin::pin] = handler;
+
+		enableInterrupts<Pin>(triggers);
+	}
+
+	template<class Pin>
+	static void
+	disconnect()
+	{
+		static_assert(Pin::port == gpioPort());
+
+		disableInterrupts<Pin>(Gpio::InputTrigger::All);
+		handlers[Pin::pin] = nullptr;
+	}
+
+private:
+	template<class Pin>
+	static void
+	enableInterrupts(Gpio::InputTrigger triggers)
+	{
+		Gpio::PortRegs<Pin::port>::enable_irq(Pin::pin, triggers);
+	}
+
+	template<class Pin>
+	static void
+	disableInterrupts(Gpio::InputTrigger triggers)
+	{
+		Gpio::PortRegs<Pin::port>::disable_irq(Pin::pin, triggers);
+	}
+
+	template<class Pin>
+	static void
+	acknowledgeInterrupts(Gpio::InputTrigger triggers)
+	{
+		Gpio::PortRegs<Pin::port>::acknowledge_irq(Pin::pin, triggers);
+	}
+
+	static void
+	irqHandler();
+	friend void MODM_ISR_NAME(IO_IRQ_BANK0)();
+	friend void MODM_ISR_NAME(IO_IRQ_QSPI)();
+
+	// In the current implementation we do not allow handlers
+	// for the same line (pin) for more than a single core.
+	static Handler handlers[Lines];
+
+	static constexpr Gpio::Port
+	gpioPort()
+	{
+%% if with_gpio
+		if constexpr (Type == IO_IRQ_BANK0_IRQn) { return Gpio::Port::Bank0; }
+%%endif
+%% if with_qspi
+		if constexpr (Type == IO_IRQ_QSPI_IRQn) { return Gpio::Port::Qspi; }
+%%endif
+		return static_cast<Gpio::Port>(-1);
+	}
+};
+
+%% if with_gpio
+using GpioIntHandler = IntHandler<IO_IRQ_BANK0_IRQn, NUM_BANK0_GPIOS>;
+%%endif
+%% if with_qspi
+using QspiIntHandler = IntHandler<IO_IRQ_QSPI_IRQn, NUM_QSPI_GPIOS>;
+%%endif
```

Comment:
Is there a way to only have one class that handles both banks? The `template<class Pin> Pin::port` already gives you the port.